DolphinQt2: use GridLayout for ControllersWindow

ligfx opened the pull request:
FormLayout is not intended for this sort of dialog:

> QFormLayout is a convenience layout class that lays out its children
> in a two-column form. The left column consists of labels and the right
> column consists of "field" widgets (line editors, spin boxes, etc.).

Before:

<img width="633" alt="screen shot 2017-05-27 at 3 23 42 am" src="https://cloud.githubusercontent.com/assets/594093/26520674/ebe9bf5e-428b-11e7-87d3-d8f187bd6f12.png">


After:
<img width="591" alt="screen shot 2017-05-27 at 3 21 42 am" src="https://cloud.githubusercontent.com/assets/594093/26520662/c130f82c-428b-11e7-9236-8a06a84af67e.png">


DolphinWX for reference:

<img width="476" alt="screen shot 2017-05-26 at 7 28 14 pm" src="https://cloud.githubusercontent.com/assets/594093/26517246/86f4f204-4249-11e7-8c84-ff7e705e9984.png">

ligfx commented:
@spycrab0 Was the intention for the indentation to match up with radio buttons?

spycrab commented:
The intention was to clarify that the options are dependent / related to the radio buttons. So that they are 'sub-options' so to speak.

leoetlino commented:
I think this was clarified a bit in WX by having proper spacing between the options. Right now, the Wii options are all kind of chained together, with no spacing between them at all.

ligfx commented:
@MayImilae @spycrab added another grid column for the indentation. Also added the vertical spacing from DolphinWX while I was in there (screenshots updated).

MayImilae commented:
That vertical spacing isn't necessary now, thanks to the indention fixing what the vertical space was trying to fix (and doing a far better job imo!), but on the just copy from wx principle, I'm fine with the vertical spacing staying for now. Up to you ligfx!

ligfx commented:
Fixed the indentation on non-macOS (something weird with Qt internals I think...).

@MayImilae I think the vertical spacing is helpful in addition to the indentation, but I don't have any HIGs or research on hand to back that up, so I've removed it. It's easy enough to change in the future if desired.
